### PR TITLE
[Merged by Bors] - feat: add a coe to ideal quotient rings

### PR DESCRIPTION
**Pull request by alexjbest**
Like we have for (Add)Subgroups already

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/Ideal/Quotient.lean [hunk 1]
```diff
@@ -104,6 +104,9 @@ def mk (I : Ideal R) : R →+* R ⧸ I where
   map_add' _ _ := rfl
 #align ideal.quotient.mk Ideal.Quotient.mk
 
+instance {I : Ideal R} : CoeTC R (R ⧸ I) :=
```
Comment (ocfnash) [inline Mathlib/RingTheory/Ideal/Quotient.lean:107]: Could you add a justification about why "Users should generally not implement this directly." does not apply here?

Comment (alexjbest) [reply Mathlib/RingTheory/Ideal/Quotient.lean:107]: TBH looking at https://github.com/leanprover/lean4/blob/master/src/Init/Coe.lean#L61 it looks to me that just a simple `Coe` is ok here and that `CoeTC` is actually more conservative in that it probably wont allow chaining if implemented directly

Comment (alexjbest) [reply Mathlib/RingTheory/Ideal/Quotient.lean:107]: So the following doesn't  work with `CoeTC` but does with `Coe`
```
variable (x : R) (I : Ideal R) (J : Ideal (R ⧸ I))
#check (x : (R⧸ I) ⧸J) 
```

**Comment by alexjbest** [comment]
I have changed both to `Coe` rather than `CoeTC` to allow chaining as this seems better, and as it goes right to left it seems loops aren't possible

**Comment by ocfnash** [comment]
Thanks!

bors merge

**Pull request closed without merging**